boards/components/src/bmp280.rs

```diff
@@ -0,0 +1,104 @@
+//! Component for the BMP280 Temperature and Pressure Sensor.
+//!
+//! Based off the SHT3x code.
+//!
+//! I2C Interface
+//!
+//! Usage
+//! -----
+//!
+//! With the default i2c address
+//! ```rust
+//! let bmp280 = components::bmp280::Bmp280Component::new(sensors_i2c_bus, mux_alarm).finalize(
+//!         components::bmp280_component_helper!(nrf52::rtc::Rtc<'static>),
+//!     );
+//! bmp280.begin_reset();
+//! ```
+//!
+//! With a specified i2c address
+//! ```rust
+//! let bmp280 = components::bmp280::Bmp280Component::new(sensors_i2c_bus, mux_alarm).finalize(
+//!         components::bmp280_component_helper!(nrf52::rtc::Rtc<'static>, capsules::bmp280::BASE_ADDR),
+//!     );
+//! bmp280.begin_reset();
+//! ```
+
+use crate::i2c::I2CComponent;
+use crate::i2c_component_helper;
+use capsules::bmp280::Bmp280;
+use capsules::virtual_alarm::{MuxAlarm, VirtualMuxAlarm};
+use capsules::virtual_i2c::MuxI2C;
+use core::mem::MaybeUninit;
+use kernel::component::Component;
+use kernel::hil::time::Alarm;
+
+use kernel::static_init_half;
+
+/// Setup static space for the driver and its requirements.
+#[macro_export]
+macro_rules! bmp280_component_helper {
+    ($A:ty) => {{
+        use capsules::bmp280;
+        $crate::bmp280_component_helper!($A, bmp280::BASE_ADDR)
+    }};
+
+    // used for specifically stating the i2c address
+    // as some boards (like nrf52) require a shift
+    ($A:ty, $address: expr) => {{
+        use capsules::bmp280::Bmp280;
+        use core::mem::MaybeUninit;
+
+        static mut BUFFER: [u8; bmp280::CALIBRATION_BYTES + 1] = [0; bmp280::CALIBRATION_BYTES + 1];
```

Comment:
Can you define an explicit buffer size in the capsule itself? Something like https://github.com/tock/tock/blob/1fe93e8dd8ab3a89ac20ff1df332681fc4ac2e29/capsules/src/spi_controller.rs#L35. That way we keep the logic for the required minimum buffer size with the driver code.

Reply:
Done.